### PR TITLE
HPCC-17907 Hpcc-init asking for password when starting Thor

### PR DESCRIPTION
**Pull request by mckellyln**
Signed-off-by: Mark Kelly <mark.kelly@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:

Install / start / stop / status of dafilesrv and hpcc-init service scripts.

### CODE REVIEW

**Comment by mckellyln** [comment]
@Michael-Gardner, @xwang2713 please review.

File: initfiles/bin/init_thorslave.in [hunk 1]
```diff
@@ -65,7 +65,11 @@ start_slaves()
 {
     # insuring dafilesrv is running on the machine as it is a prerequisite
     dafilesrv_owner=$(stat -c "%U" ${INIT_PATH}/dafilesrv 2>/dev/null || stat -f %Su ${INIT_PATH}/dafilesrv 2>/dev/null)
-    [[ "${dafilesrv_owner}" != "${user}" ]] && cmd_prefix="sudo"
+    # what is stat -f %Su and why check file owner ?
```
Comment (mckellyln) [inline initfiles/bin/init_thorslave.in]: @Michael-Gardner do you recall the answers to these questions ?

Comment (Michael-Gardner) [inline initfiles/bin/init_thorslave.in]: It's part of an osx compatibility patch.  See https://github.com/hpcc-systems/HPCC-Platform/commit/007b784d062a859181e2aacf50237f7111251649

**Review by Michael-Gardner — APPROVED** [review]

**Comment by Michael-Gardner** [comment]
@mckellyln Looks good.

**Comment by mckellyln** [comment]
@Michael-Gardner removed comment, rebased.

**Pull request merged**